File: MoveZeroes/MoveZeroes.py
class Solution(object):
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: void Do not return anything, modify nums in-place instead.
        """
        
        zero_count = 0
        for i in xrange(len(nums)):
            if nums[i] == 0:
                zero_count += 1
                continue
            nums[i - zero_count] = nums[i]
        for i in xrange(zero_count, 0, -1):
            nums[-i] = 0

File: MoveZeroes/MoveZeroes.cpp
#include<vector>
using namespace std;

class Solution {
public:
	void moveZeroes(vector<int>& nums) {
		int zero_count = 0;
		int len = nums.size();
		for (int i = 0; i < len; i++) {
			if (nums[i] == 0) {
				zero_count++;
				continue;
			}
			nums[i - zero_count] = nums[i];
		}
		for (int i = 0; i < zero_count; i++) {
			nums[len - i - 1] = 0;
		}
	}
};